From 3dc4c19aa4c72987b98d9d8237ee082b4c96c6ee Mon Sep 17 00:00:00 2001
From: Alexander Couzens <lynxis@c-base.org>
Date: Sun, 1 Apr 2012 13:36:52 +0300
Subject: [PATCH 65/75] Fix potential double free and use of freed memory in
 RADIUS client

ieee802_1x_encapsulate_radius() frees the RADIUS message if
radius_client_send() returns error. This could have resulted in use of
freed memory and double freeing of the RADIUS message if send() fails
since the message is also left in the retransmit list. Avoid this by not
returning error to the caller in such a case.

Signed-off-by: Alexander Couzens <lynxis@c-base.org>
(cherry picked from commit a4e73a0e470cde6fcf3983aca277f01540bb37fa)
---
 src/radius/radius_client.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/src/radius/radius_client.c
+++ b/src/radius/radius_client.c
@@ -684,7 +684,7 @@ int radius_client_send(struct radius_cli
 	radius_client_list_add(radius, msg, msg_type, shared_secret,
 			       shared_secret_len, addr);
 
-	return res;
+	return 0;
 }
 
 
